packages/amplify/amplify_flutter/example/integration_test/config_test.dart

```diff
@@ -154,141 +153,72 @@ void main() {
-  final allPlugins = <AmplifyPluginInterface>[
-    AmplifyAuthCognito(),
-    AmplifyAnalyticsPinpoint(),
-    AmplifyAPI(),
-    dataStorePlugin,
-    AmplifyStorageS3(),
-  ];
-
-  /// Calls `Amplify.reset` on the native library
-  Future<void> resetAmplify() {
-    const methodChannel = MethodChannel('amplify_flutter/integ_tests');
-    return methodChannel.invokeMethod<void>('reset');
-  }
-
-  /// Simulates a hot reload
-  void hotReload() {}
-
-  /// Simulates a hot restart
-  void hotRestart() {
-    Amplify = MethodChannelAmplify();
-    Amplify.Analytics.plugins.clear();
-    Amplify.Auth.plugins.clear();
-    Amplify.API.plugins.clear();
-    Amplify.DataStore.plugins.clear();
-    Amplify.Storage.plugins.clear();
+  final isMobile = !kIsWeb && (Platform.isIOS || Platform.isAndroid);
+
+  final mobilePlugins = <AmplifyPluginInterface>[];
+
+  /// Add plugins currently supported on mobile only
+  if (isMobile) {
+    mobilePlugins.addAll([
+      AmplifyDataStore(modelProvider: modelProvider),
+      AmplifyAnalyticsPinpoint(),
+      AmplifyAPI(),
+      AmplifyStorageS3(),
+    ]);
   }
 
   group(
     'Configuration',
     () {
-      setUp(() async {
-        await resetAmplify();
-        hotRestart();
-      });
+      testWidgets(
+        'addPlugin registers a single plugin',
+        (WidgetTester tester) async {
+          expect(Amplify.Auth.plugins, isEmpty);
+          await expectLater(Amplify.addPlugin(AmplifyAuthCognito()), completes);
+          expect(Amplify.Auth.plugins, isNotEmpty);
+        },
+      );
+
+      testWidgets(
+        'addPlugins registers multiple plugins',
+        (WidgetTester tester) async {
+          expect(Amplify.Analytics.plugins, isEmpty);
+          expect(Amplify.API.plugins, isEmpty);
+          expect(Amplify.DataStore.plugins, isEmpty);
+          expect(Amplify.Storage.plugins, isEmpty);
+          await expectLater(Amplify.addPlugins(mobilePlugins), completes);
+          expect(Amplify.Analytics.plugins, isNotEmpty);
+          expect(Amplify.API.plugins, isNotEmpty);
+          expect(Amplify.DataStore.plugins, isNotEmpty);
+          expect(Amplify.Storage.plugins, isNotEmpty);
+        },
+        skip: !isMobile,
+      );
+
+      testWidgets(
+        'amplify configures successfully',
+        (WidgetTester tester) async {
+          await expectLater(Amplify.configure(amplifyconfig), completes);
+          expect(Amplify.isConfigured, true);
+        },
+      );
 
-      testWidgets('multiple configure calls throws',
+      testWidgets('multiple configure calls throw',
           (WidgetTester tester) async {
-        await expectLater(Amplify.configure(amplifyconfig), completes);
-        expect(Amplify.isConfigured, isTrue);
-
-        await expectLater(
-          Amplify.configure(amplifyconfig),
-          throwsAlreadyConfigured,
-        );
-      });
-
-      testWidgets('hot reload persists all state', (WidgetTester tester) async {
```

Comment:
Is the idea behind removing the hot restart / hot reload tests that these will not be needed in once the native plugins are no longer used and they are not worth maintaining?

Reply:
Yeah, and will probably be replaced with Amplify.reset tests.